_pytest/recwarn.py

```diff
@@ -28,9 +28,18 @@ def pytest_namespace():
             'warns': warns}
 
 
-def deprecated_call(func, *args, **kwargs):
+def deprecated_call(func=None, *args, **kwargs):
     """Assert that ``func(*args, **kwargs)`` triggers a DeprecationWarning.
+
+    This function can be used as a context manager::
+
+        >>> with deprecated_call():
+        ...    myobject.deprecated_method()
     """
+    if not func:
+        warnings.simplefilter('always')
```

Comment:
Ah, I've found that, currently am trying to figure out why it only works if append=True is unset:
`WarningsRecorder.__enter__:warnings.simplefilter('always')`   - works fine
`WarningsRecorder.__enter__:warnings.simplefilter('always', append=True)`   - the existing code, breaks my PR tests